pkg/alertmanager/alertmanager.go

```diff
@@ -433,9 +432,9 @@ func clusterWait(position func() int, timeout time.Duration) func() time.Duratio
 
 // ApplyConfig applies a new configuration to an Alertmanager.
 func (am *Alertmanager) ApplyConfig(conf *definition.PostableApiAlertingConfig, tmpls []alertingTemplates.TemplateDefinition, rawCfg string, tmplExternalURL *url.URL, staticHeaders map[string]string) error {
-	templates := make([]io.Reader, 0, len(tmpls))
+	templates := make([]string, 0, len(tmpls))
 	for _, tmpl := range tmpls {
-		templates = append(templates, strings.NewReader(tmpl.Template))
+		templates = append(templates, tmpl.Template)
```

Comment:
Then here you could just do the below and probably (not tested 😅 ) not change anything else:
```suggestion
		templates = append(templates, &unlimitedStringReader{tmpl.Template})
```